src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+        }
+
+        // find discrete intervals that contain the breakpoint evidence
+        final Iterator<Interval> intervalItr =
+                evidenceRDD
+                        .mapPartitions(evidenceItr ->
+                                new MapPartitioner<>(evidenceItr,
+                                        new IntervalMapper(maxFragmentSize), new BreakpointEvidence(nContigs)), true)
+                        .collect()
+                        .iterator();
+
+        // coalesce overlapping intervals (can happen at partition boundaries)
+        final List<Interval> intervals = new ArrayList<>();
+        if ( intervalItr.hasNext() ) {
+            Interval prev = intervalItr.next();
+            while ( intervalItr.hasNext() ) {
+                final Interval next = intervalItr.next();
+                if ( prev.isDisjointFrom(next) ) {
+                    intervals.add(prev);
+                    prev = next;
+                } else {
+                    prev = prev.join(next);
+                }
+            }
+            intervals.add(prev);
+        }
+
+        if ( locations.evidenceDir != null ) evidenceRDD.unpersist();
+
+        // record the intervals
+        if ( locations.intervalFile != null ) {
+            try (final OutputStreamWriter writer = new OutputStreamWriter(new BufferedOutputStream(
+                    BucketUtils.createFile(locations.intervalFile, pipelineOptions)))) {
+                for (final Interval interval : intervals) {
+                    final String seqName = contigs.get(interval.getContig()).getSequenceName();
+                    writer.write(seqName + " " + interval.getStart() + " " + interval.getEnd() + "\n");
+                }
+            } catch (final IOException ioe) {
+                throw new GATKException("Can't write intervals file " + locations.intervalFile, ioe);
+            }
+        }
+
+        return intervals;
+    }
+
+    /** gather some interesting factoids about the reads in aggregate */
+    @VisibleForTesting static ReadMetadata getMetadata( final SAMFileHeader header,
+                                             final int meanBasesPerTemplate ) {
+        final List<SAMReadGroupRecord> groups = header.getReadGroups();
+        final int nGroups = groups.size();
+        final ReadMetadata.ReadGroupFragmentStatistics groupStats =
+                //TODO: get real data
+                new ReadMetadata.ReadGroupFragmentStatistics(400.f, 75.f);
+        final List<ReadMetadata.ReadGroupFragmentStatistics> stats = new ArrayList<>(nGroups);
+        for ( int idx = 0; idx != nGroups; ++idx ) {
+            stats.add(groupStats);
+        }
+        return new ReadMetadata(header, stats, groupStats, meanBasesPerTemplate);
+    }
+
+    /** mean number of sequenced bases per template (i.e., read length if unpaired, twice the read length if paired) */
+    @VisibleForTesting static int getMeanBasesPerTemplate(final JavaRDD<GATKRead> reads ) {
```

Comment:
Yes I think `sample` runs over the whole RDD. I might just switch to `take` and see if that speeds things up. In the vast majority of cases there will only be one read length anyway, aside from hard clipping which we shouldn't have to worry about too much.

Another option would be to take it as an (optional) parameter, or take a path to the picard metrics file that has that info and just parse it out of there.

Just trying to think of ways to cut down on the pipeline runtime.